extensions/interactions/AlgebraicExpressionInput/protractor.js

```diff
@@ -17,10 +17,29 @@
  * interaction
  */
 
+var action = require(process.cwd() + '/core/tests/protractor_utils/action.js');
 var objects = require(process.cwd() + '/extensions/objects/protractor.js');
+var waitFor = require(
+  process.cwd() + '/core/tests/protractor_utils/waitFor.js');
 
-var customizeInteraction = function() {
-  // There are no customizations.
+var customizeInteraction = async function(elem, customLetters) {
+  await waitFor.presenceOf(elem.element(by.css(
+    '.protractor-test-custom-letters-div')), 'Element took too long to load.');
+  for (let letter of customLetters) {
+    await action.click('Math OSK Tab', elem.element(by.buttonText('abc')));
+    let letterIsPresent = true;
+    try {
+      await browser.wait(
+        await waitFor.until.presenceOf(elem.element(by.buttonText(letter))),
+        waitFor.DEFAULT_WAIT_TIME_MSECS, 'Element took too long to load.');
```

Comment:
Could you dig into why just calling the waitFor function doesn't work? Copying the function code is really not ideal

Reply:
I did some more experimenting and it seems like calling the function doesn't explicitly throw an exception which is why the catch block is never being reached. Now, I tried adding a `throw new Error(errorMessage);` in the `presenceOf` function and then it threw exceptions. But, having this would mean that we would have to wrap all invocations of the `presenceOf` function in a try-catch block, otherwise, they will throw errors in the absence of said element. Also, I don't see such a pattern of catching errors from waitFor functions elsewhere in the codebase.

Reply:
Update: I've used a much simpler approach (thanks @aks681!), which makes this problem obsolete.

Reply:
@iamprayush Here's an example where we use a similar pattern:

https://github.com/oppia/oppia/blob/develop/core/tests/protractor_utils/TopicsAndSkillsDashboardPage.js#L342

I'll also dig into this some more on my end to see if I can figure out why it isn't working for you

Reply:
I see. I tried this pattern (I also called the click function inside the try block and it worked).
```javascript
for (let letter of customLetters) {
    await action.click('Math OSK Tab', elem.element(by.buttonText('abc')));
    try {
      await waitFor.presenceOf(
        elem.element(by.buttonText(letter)),
        'Letter took too long to be present.');
      await elem.element(by.buttonText(letter)).click();
    } catch (e) {
      await action.click('Math OSK Tab', elem.element(by.buttonText('αβγ')));
    }
    await action.click('Math OSK Letter', elem.element(by.buttonText(letter)));
  }
```